From ec852e6d691f6da92e51661e628aee67f42141f7 Mon Sep 17 00:00:00 2001
Message-Id: <patch-3.4-ec852e6d691-20220218T205753Z-avarab@gmail.com>
In-Reply-To: <cover-0.4-00000000000-20220218T205753Z-avarab@gmail.com>
References: <cover-0.4-00000000000-20220218T205753Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Mon, 13 Dec 2021 01:08:17 +0100
Subject: [PATCH 3/4] test-lib: add "strip_path_prefix" to XSAN_OPTIONS
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Change the stack traces emitted by the tests under LSAN and ASAN from
showing stack frames like:

    #0 0x4323b8 in malloc (/home/avar/g/git/t/helper/test-tool+0x4323b8)

To strip out the path to the build directory, so we'll instead emit:

    #0 0x4323b8 in malloc (t/helper/test-tool+0x4323b8)

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 t/test-lib.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/t/test-lib.sh b/t/test-lib.sh
index 4f523b82ce5..acf03e461d7 100644
--- a/t/test-lib.sh
+++ b/t/test-lib.sh
@@ -50,6 +50,7 @@ prepend_var () {
 
 # If [AL]SAN is in effect we want to abort so that we notice problems.
 prepend_var XSAN_OPTIONS : abort_on_error=1
+prepend_var XSAN_OPTIONS : strip_path_prefix=\"$GIT_BUILD_DIR/\"
 
 # If we were built with ASAN, it may complain about leaks
 # of program-lifetime variables. Disable it by default to lower
-- 
2.35.1.1031.g277d4562d2e

